src/core/server/ui_settings/types.ts

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { SavedObjectsClientContract, SavedObjectAttribute } from '../saved_objects/types';
+/**
+ * Client that provides access to the UiSettings stored in elasticsearch.
+ *
+ * @public
+ */
+export interface IUiSettingsClient {
```

Comment:
We don't currently expose any observables here. Should we to encourage consumers to adapt to changing values? I'm not sure exactly how / where this is used on the server. Maybe it only makes sense in the context of a request so they won't change anyways?

Reply:
we do provide UiSettingsClient via request handler context https://github.com/elastic/kibana/pull/48413/files/d5a0b8b7832393b9778337b5c2c3f1b8bde59926#diff-8fa226a3839a3775ca15bdb3a2a28ffaR19

>  We don't currently expose any observables here

We don't store value in memory, but delegate to **savedObjects**. I don't think we can use observables here - values can be set via savedObjects API, from other kibana instances, etc.

Reply:
Makes sense 👍